week3/class_version/cv_tictactoe_board.py

```diff
@@ -0,0 +1,207 @@
+"""
+Module sets up the TicTacToeBoard class that can be used
+to play tictactoe
+"""
+
+EMPTY = 0
+PLAYERX = 1
+PLAYERO = 2
+DRAW = -2
+
+def switch_player(player):
+    if player == PLAYERO:
+        return PLAYERX
+    return PLAYERO
+
+class TicTacToeBoard(object):
+    """
+    Class to represent a Tic-Tac-Toe board.
+    """
+
+    def __init__(self, dimension, reverse=False, board=None):
+        """
+        Initialize the TTTBoard object with the given dimension and
+        whether or not the game should be reversed.
+        """
+        self._dimension = dimension
+        self._reverse = reverse
+        self._board = self.new_board()
+
+    def __str__(self):
+        """
+        Human readable representation of the board.
+        """
+        dashes = '---'*self.dimension + '-'*(self.dimension-1)
+
+        replacements = [('[[', ' '), ('[', ''),
+                        ('],', '\n'+dashes+'\n'), (', ', ' | '),
+                        (']]', ''), ('0', ' '),
+                        ('1', 'X'), ('2', 'O')]
+
+        board_string = str(self.board)
+        for index in replacements:
+            board_string = board_string.replace(index[0], index[1])
+
+        return board_string
+
+    @property
+    def dimension(self):
+        """
+        Return the dimension of the board.
+        """
+        return self._dimension
+
+    @dimension.setter
+    def dimension(self, value):
+        """
+        Set the value of dim
+        """
+        self._dimension = value
+
+    def get_dim(self):
+        """
+        Method used for class version of monte carlo tictactoe to 
+        get the dimension of the board
+        """
+        return self.dimension
+
+    @property
+    def board(self):
+        """
+        Return the board
+        """
+        return self._board
+
+    @board.setter
+    def board(self, value):
+        """
+        Set the value of the board
+        """
+        self._board = value
+
+    def new_board(self):
+        """
+        Generates a board of size dimension with 0's as values
+        """
+        return [[0 for dummy_col in range(self.dimension)]
+                for dummy_row in range(self.dimension)]
+
+    def square(self, row, col):
+        """
+        Returns one of the three constants EMPTY, PLAYERX, or PLAYERO
+        that correspond to the contents of the board at position (row, col).
+        """
+        if self.board[row][col] == 1:
+            return PLAYERX
+        elif self.board[row][col] == 2:
+            return PLAYERO
+        return EMPTY
```

Comment:
This could be simpler
```
return self.board[row][col]
```
A couple things:
 * You define the constants above. You should reuse them here: 
```
if self.board[row][col] == PLAYERO:
  return PLAYERO
```
But, since you're just returning what's there, just return it and forget the equality checks. :)